File: 2-Basic-Mathematical-Knowledge-for-Algorithms/008-Brute_Force_1/javascript/main.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.on('line', (line) => {
    // 1. 入力処理
    const [n, s] = line.trim().split(' ').map(Number);

    // 2. 最適化された計算
    let count = 0;
    for (let r = 1; r <= n; r++) {
        const maxB = Math.min(n, s - r);
        count += Math.max(0, maxB);
    }

    // 3. 出力処理
    console.log(count);
    
    rl.close();
});

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/008-Brute_Force_1/python/main.py
# 1. 入力処理
n, s = map(int, input().split())

# 2. 最適化された計算
count = 0
for r in range(1, n + 1):
    max_b = min(n, s - r)
    count += max(0, max_b)

# 3. 出力処理
print(count)